components/style/stylist.rs

```diff
@@ -269,9 +265,23 @@ impl Stylist {
         self.selectors_for_cache_revalidation.len()
     }
 
-    /// Gets a reference to the invalidation map.
-    pub fn invalidation_map(&self) -> &InvalidationMap {
-        &self.invalidation_map
+    /// Returns the number of entries in invalidation maps.
+    pub fn num_invalidations(&self) -> usize {
+        self.cascade_data.iter_origins()
+            .map(|d| d.invalidation_map.len()).sum()
```

Comment:
nit: I'd move `sum`, and maybe `iter_origins`, to its own line, but no huge deal.